protocol/x/clob/e2e/isolated_subaccount_orders_test.go

```diff
@@ -0,0 +1,494 @@
+				PlaceOrder_Alice_Num0_Id0_Clob3_Buy_1ISO_Price10_GTB5,
+				PlaceOrder_Bob_Num0_Id0_Clob3_Sell_1ISO_Price10_GTB5,
+			},
+			collateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 30_000_000_000, // $30,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 5_000_000_000, // $5,000 USDC
+			},
+			expectedOrdersFilled: []clobtypes.OrderId{
+				PlaceOrder_Alice_Num0_Id0_Clob3_Buy_1ISO_Price10_GTB5.Order.OrderId,
+				PlaceOrder_Bob_Num0_Id0_Clob3_Sell_1ISO_Price10_GTB5.Order.OrderId,
+			},
+			expectedSubaccounts: []satypes.Subaccount{
+				Alice_Num0_IsolatedAfterMatch,
+				Bob_Num0_IsolatedAfterMatch,
+			},
+			expectedCollateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 10_000_000_000, // $30,000 USDC - $10,000 USDC - $10,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 24_999_999_999, // $5,000 USDC + $10,000 USDC + $10,000 USDC - fee (1 quote quantum)
+			},
+		},
+		"Isolated subaccount closing position moves collateral back to cross collateral pool": {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_1ISO_LONG_10_000USD,
+				constants.Bob_Num0_1ISO_LONG_10_000USD,
+			},
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+				constants.EthUsd_20PercentInitial_10PercentMaintenance,
+				constants.IsoUsd_IsolatedMarket,
+			},
+			clobPairs: []clobtypes.ClobPair{
+				constants.ClobPair_Btc,
+				constants.ClobPair_Eth,
+				constants.ClobPair_3_Iso,
+			},
+			// Orders should match.
+			orders: []clobtypes.MsgPlaceOrder{
+				PlaceOrder_Alice_Num0_Id0_Clob3_Buy_1ISO_Price10_GTB5,
+				PlaceOrder_Bob_Num0_Id0_Clob3_Sell_1ISO_Price10_GTB5,
+			},
+			collateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 10_000_000_000, // $10,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedOrdersFilled: []clobtypes.OrderId{},
+			expectedSubaccounts: []satypes.Subaccount{
+				Alice_Num0_MoreIsolatedAfterMatch,
+				Bob_Num0_CrossAfterMatch,
+			},
+			// No changes as no match should have happened.
+			expectedCollateralPoolBalances: map[string]int64{
+				// $10,000 USDC + $10,000 USDC + (match) 100 quote quantums - fee (1 quote quantum)
+				satypes.ModuleAddress.String(): 20_000_000_099,
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+					// $30,000 USDC - $10,000 USDC - (match) 100 quote quantums
+				).String(): 19_999_999_900,
+			},
+		},
+		`Empty subaccount will not have matches for isolated market orders if cross collateral pool does not 
+		have enough balance`: {
```

Comment:
for my understanding - this is something that can potentially happen? why would cross collat pool not have enough balance?

Reply:
attacks from -ve TNC / bypassing withdrawal gating?

Reply:
Yes, exactly.